Lib/idlelib/idle_test/test_parenclose.py

```diff
@@ -0,0 +1,97 @@
+'''Test idlelib.parenclose.
```

Comment:
Add MockEvent and ParenCloseTest to test_parenmatch.  I have not reviewed in detail, but have a couple of comments.